Snakefile

```diff
@@ -0,0 +1,402 @@
+        additional_info = "data/gisaid/additional_info.tsv"
+    shell:
+        """
+        ./bin/transform-gisaid {input.ndjson} \
+            --output-metadata {output.metadata} \
+            --output-fasta {output.fasta}  \
+            --output-additional-info {output.additional_info} \
+            --output-unix-newline > {output.flagged_annotations};
+        """
+
+rule download_nextclade:
+    params:
+        dst_source=lambda w: config["s3_dst"] + '/nextclade.tsv.gz',
+        src_source=lambda w: config["s3_src"] + '/nextclade.tsv.gz',
+    output:
+        nextclade = "data/{database}/nextclade_old.tsv"
+    shell:
+        """
+        ./bin/download-from-s3 {params.dst_source} {output.nextclade} ||  \
+        ./bin/download-from-s3 {params.src_source} {output.nextclade}
+        """
+
+rule get_sequences_without_nextclade_annotations:
+    """Find sequences in FASTA which don't have clades assigned yet"""
+    input:
+        fasta = "data/{database}/sequences.fasta",
+        nextclade = "data/{database}/nextclade_old.tsv",
+    output:
+        fasta = "data/{database}/nextclade.sequences.fasta"
+    shell:
+        """
+        ./bin/filter-fasta \
+            --input_fasta={input.fasta} \
+            --input_tsv={input.nextclade} \
+            --output_fasta={output.fasta} \
+        """
+
+rule annotate_via_nextclade:
+    message:
+        """
+        If there are sequences without clades, then run nextclade to assign
+        clades and calculate other useful metrics.
+        """
+    input:
+        sequences = "data/{database}/nextclade.sequences.fasta",
+        nextclade_info = "data/{database}/nextclade_old.tsv"
+    params:
+        new_info = temp("data/{database}/nextclade_new.tsv"),
+        nextclade_input_dir = temp(directory("data/{database}/nextclade_inputs")),
+        nextclade_output_dir = temp(directory("data/{database}/nextclade"))
+    threads: 8
+    output:
+        nextclade_info = "data/{database}/nextclade.tsv"
+
+    ## todo - move this code into a shell script / expand the abilities of `./bin/run-nextclade`
+    ## note - this conditionality on a non-empty input fasta was GISAID only, but it makes sense for GenBank too
+    shell:
+        """
+        if [ ! -s {input.sequences:q} ]; then
+            echo "[ INFO] No new sequences for Nextclade to process. Skipping. (rule find_clades_via_nextclade)"
+            mv {input.nextclade_info} {output.nextclade_info}
+        else
+            ./bin/run-nextclade \
+                {input.sequences:q} \
+                {params.new_info} \
+                {params.nextclade_input_dir} \
+                {params.nextclade_output_dir}
+            # Join new and old clades, so that next run won't need to process sequences that are already processed
+            ./bin/join-rows \
+                {input.nextclade_info:q} \
+                {params.new_info} \
+                -o {output.nextclade_info:q}
+        fi
```

Comment:
Just making a note that we should consider moving this conditional logic either a) _into_ a single script that handles the logic or b) _out of_ a single Snakemake rule and up to a workflow-level conditional statement. I've personally had more success debugging logic with the first approach where the I can test a single script manually outside of the workflow and inside the workflow.

Reply:
👍  resolving as this todo in #239 (and there's also a todo comment in the code for this rule)